doc/docs/api.md

```diff
@@ -341,3 +341,21 @@ ioFiles: cats.effect.IO[Seq[java.io.File]]
 
 Note that this example requires the `coursier-cats-interop` module
```

Comment:
I think this whole page could use a bit of a reorg, e.g. there's some duplication around the whole future/io part. I'd extract that to a separate section (outside Fetch and Resolve). What do you think?